app/full_text_search.py

```diff
@@ -42,15 +56,34 @@ def search(repo, query_txt, max_results):
     results = []
     if not query_txt:
         return results
-    index = appengine_search.Index(name=PERSON_FULL_TEXT_INDEX_NAME)
-    query_txt += ' AND (repo: ' + repo + ')'
+
+    query_txt = re.sub('|:|OR|AND|"', '', query_txt)
```

Comment:
How about putting each word into quote e.g., [hoge OR foo] -> ["hoge" "OR" "foo"]? Will it work? For double-quotes, you may just remove them (as you already do) or somehow escape them.

Also, please add a comment why you do this.